QA/PoroDevTest/src/main/java/common/ui_setup/pom_setup/PoroDevPom/LoginPage.java

```diff
@@ -1,55 +1,74 @@
         super(driver);
-        driver.get(SetupConstants.BASE_URL);
+    }
+
+    public LoginPage(WebDriver driver, String url) {
+        super(driver);
+        driver.get(url);
     }
 
     //element section
-    @FindBy(id = "basic_email")
-    WebElement emailField;
-    @FindBy(id = "basic_password")
-    WebElement passField;
+    @FindBy(name = "email")
+    WebElement we_emailInputField;
+    @FindBy(name = "password")
+    WebElement we_passInputField;
     @FindBy(className = "ant-btn-primary")
-    WebElement loginButton;
+    WebElement we_loginButton;
     //ASSERTION ELEMENTS
-    @FindBy(xpath = "//div[contains(text(),'Please input your username')]")
-    public WebElement emailErrorMsg;
-    @FindBy(xpath = "//div[contains(text(),'Please input your password')]")
-    public WebElement passErrorMsg;
-    @FindBy(xpath = "//span[contains(text(),'Request failed with status code 401')]")
-    public WebElement errorAfterBadCredentials;
-    @FindBy(className = "hvVmBA")
-    public WebElement userFirstNameAfterLogin;
+    @FindBy(xpath = "//span[contains(text(),'Email is required')]")
+    WebElement we_email_requiredMsg;
+    @FindBy(xpath = "//span[contains(text(),'Password is required')]")
+    WebElement we_pass_requiredMsg;
+    @FindBy(xpath = "//span[contains(text(),'Invalid email')]")
+    WebElement we_email_invalidMsg;
+    @FindBy(xpath = "//span[contains(text(),'Wrong password')]")
+    WebElement we_pass_invalidMsg;
     @FindBy(xpath = "//span[contains(text(),'Successful login')]")
-    public WebElement successfulLogin;
+    WebElement we_successfulLogin;
+    @FindBy(className = "ant-message")
+    WebElement we_badRequestError;
 
-    //methods section
-    public void enterEmail(String email) {
-        elementControl.sendText(emailField, email);
+    //functional methods section
+
+    public void login(String email, String password) throws InterruptedException {
+        BasePage.sendText(we_emailInputField,email);
+        BasePage.sendText(we_passInputField, password);
+        BasePage.clickElement(we_loginButton);
+        Thread.sleep(2000);
     }
 
-    public void enterPassword(String pass) {
-        elementControl.sendText(passField, pass);
+    //ASSERT METHODS
+    public void assert_valid_login(String expectedResults) {
+        Assert.assertEquals(BasePage.getTextFromElement(we_successfulLogin), expectedResults);
     }
 
-    public void clickLoginButton() {
-        elementControl.clickElement(loginButton);
+    public void assert_login_with_invalid_form_password(String expectedResults) {
+        Assert.assertEquals(BasePage.getTextFromElement(we_pass_invalidMsg), expectedResults);
     }
 
-    public void login_and_assert(String email, String password) throws InterruptedException {
-        enterEmail(email);
-        enterPassword(password);
-        clickLoginButton();
-        Thread.sleep(2000);
+    public void assert_login_with_invalid_form_email(String expectedResults) {
```

Comment:
If you are getting valid/invalid message from the same element, please use one method, and you can put a switch there, or go even without anything, with printing the values of expectedResult/actualResult.